### ISSUE TITLE
What's the main difference between this version and the first version?

### ISSUE THREAD

**Issue by Kelvinson** [gaoxiang12/slambook2#4]
Hi Dr Gao, 

Thanks for your efforts to make your work(tutorials and code etc) public, I am really looking forward to your new book. As the title, what's the main difference between this version and the first version? Thank you very much!

**Comment by gaoxiang12** [comment]
Hi @Kelvinson ,
In edition 2, we have made a lot of changes based on the first version, including:
- Deeper understanding of the algorithms. We will implement most the cv algorithms based on raw cpp code instead of concatenating opencv functions. For example we have a self-implemented ORB detector [here](https://github.com/gaoxiang12/slambook2/blob/master/ch7/orb_self.cpp): and a optical flow [here](https://github.com/gaoxiang12/slambook2/blob/master/ch8/optical_flow.cpp).
- More complete project: We have a complete stereo VO system implemented in <1k lines in our project chapter.
- Fix a bunch of typos, bugs and errors in the first version.

I've spent serveral months on this version and hope you like it!
